fhevm/tee_comparison.go

```diff
@@ -0,0 +1,227 @@
+package fhevm
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/zama-ai/fhevm-go/fhevm/tfhe"
+	"github.com/zama-ai/fhevm-go/tee"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func doComparison1Operation(op string, environment EVMEnvironment, caller common.Address, input []byte, runSpan trace.Span, operator func(uint64, uint64) bool) ([]byte, error) {
+	logger := environment.GetLogger()
+
+	lp, rp, lhs, rhs, err := extract2Operands(op, environment, input, runSpan)
+	if err != nil {
+		logger.Error(op, "failed", "err", err)
+		return nil, err
+	}
+
+	// If we are doing gas estimation, skip execution and insert a random ciphertext as a result.
+	if !environment.IsCommitting() && !environment.IsEthCall() {
+		return importRandomCiphertext(environment, lhs.fheUintType()), nil
+	}
+
+	// TODO ref: https://github.com/Inco-fhevm/inco-monorepo/issues/6
+	if lp.FheUintType == tfhe.FheUint128 || lp.FheUintType == tfhe.FheUint160 {
+		panic("TODO implement me")
+	}
+
+	// Using math/big here to make code more readable.
+	// A more efficient way would be to use binary.BigEndian.UintXX().
+	// However, that would require a switch case. We prefer for now to use
+	// big.Int as a one-liner that can handle variable-length bytes.
+	//
+	// Note that we do arithmetic operations on uint64, then we convert th
+	// result back to the FheUintType.
+	l := big.NewInt(0).SetBytes(lp.Value).Uint64()
+	r := big.NewInt(0).SetBytes(rp.Value).Uint64()
+
+	result := operator(l, r)
+	resultBz, err := marshalBool(result)
+	if err != nil {
+		return nil, err
+	}
+	teePlaintext := tee.NewTeePlaintext(resultBz, 0, caller)
```

Comment:
Still a lot of repeated code. Could you see if we can factorize that? Also factorize with arithmetic operations

Reply:
In general, a good rule is: if you repeat 2 times, it's okay. If you repeat 3+ times, factorize into a function.

Reply:
Okay.

Reply:
Let me know if you find something. I can also help take a look at factorizing if you can't find anything.